shared/src/main/scala/drt/shared/Api.scala

```diff
@@ -285,38 +280,61 @@ trait WithTerminal[A] extends Ordered[A] {
   def terminal: Terminal
 }
 
-case class UniqueArrival(number: Int, terminal: Terminal, scheduled: MillisSinceEpoch)
-  extends WithLegacyUniqueId[Int, UniqueArrival]
-    with WithTimeAccessor
-    with WithTerminal[UniqueArrival] {
+sealed trait UniqueArrivalLike {
+  val number: Int
+  val terminal: Terminal
+  val scheduled: MillisSinceEpoch
+}
+
+case class LegacyUniqueArrival(number: Int, terminal: Terminal, scheduled: MillisSinceEpoch) extends UniqueArrivalLike
+
+object LegacyUniqueArrival {
```

Comment:
do we want to add @deprecated, just to more explicit about not using it for new development ?

Reply:
I'm not sure about that. It's not something we can remove because it's required for use with previously persisted data. I would read @deprecated as meaning we will be removing it.